packages/datadog_session_replay/lib/src/capture/recorder.dart

```diff
@@ -0,0 +1,196 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2025-Present Datadog, Inc.
+
+import 'package:datadog_flutter_plugin/datadog_internal.dart';
+import 'package:flutter/widgets.dart';
+
+import '../datadog_session_replay_platform_interface.dart';
+import '../rum_context.dart';
+import '../widgets.dart';
+import 'capture_node.dart';
+import 'element_recorders/container_recorder.dart';
+import 'pointer_capture.dart';
+import 'view_tree_snapshot.dart';
+
+abstract interface class ElementRecorder {
+  CaptureNodeSemantics? captureSemantics(
+    Element element,
+    CapturedViewAttributes attributes,
+  );
+}
+
+class KeyGenerator {
+  // This is close to JavaScript's MAX_SAFE_INT (53-bit)
+  static const int maxKey = 0x20000000000000;
```

Comment:
I recall that we had issues when going beyond 32-bit

On iOS we have limited the max value of IDs to Int32
https://github.com/DataDog/dd-sdk-ios/blob/9a80e2eccab6f69999bf75ffb77b0ec52d8b0392/DatadogSessionReplay/Sources/Recorder/ViewTreeSnapshotProducer/ViewTreeSnapshot/NodeIDGenerator.swift#L30

On iOS, it's incremental ID generation, but I think it's fine to randomize as long as we can keep it consistent for the same view.

 On Android it's taking the memory address and trims it down to 32-bits.